poc_entropy/src/server.rs

```diff
@@ -8,36 +11,61 @@ use axum::{
     Json, Router,
 };
 use chrono::Utc;
+use helium_proto::{
+    services::poc_entropy::{EntropyReqV1, PocEntropy, Server as GrpcServer},
+    EntropyReportV1,
+};
 use std::net::SocketAddr;
 use tokio::{sync::watch, time::Duration};
-use tower_http::trace::TraceLayer;
+
+struct EntropyServer {
+    entropy_watch: MessageReceiver,
+}
+
+#[tonic::async_trait]
+impl PocEntropy for EntropyServer {
+    async fn entropy(
+        &self,
+        _request: tonic::Request<EntropyReqV1>,
+    ) -> Result<tonic::Response<EntropyReportV1>, tonic::Status> {
+        let entropy = &*self.entropy_watch.borrow();
```

Comment:
This is needed because the result of `borrow()` is a Watch which needs to be deref'd using `*` but then we borrow it because we don't actually need ownership of it

Reply:
so you're 1) borrowing the watcher, 2) deref'ing the watcher to get the value of the message it's retaining and 3) taking a reference to the message or the entropy value it contains, and the `.borrow()` method is for the compiler to distinguish taking a reference to the watcher from the reference to the entropy data you're retrieving. do i have that right?

Reply:
The semantics of [sync::watch::Receiver](https://docs.rs/tokio-sync/latest/tokio_sync/watch/struct.Receiver.html) are that `borrow()` puts a lock on the value held in the watch.. The deref gets though that lock to the contained value